Formula/driftctl.rb

```diff
@@ -0,0 +1,21 @@
+class Driftctl < Formula
+  desc "Detect, track and alert on infrastructure drift"
+  homepage "https://driftctl.com"
+  url "https://github.com/cloudskiff/driftctl/archive/v0.2.2.tar.gz"
+  sha256 "22ec519af467819dc0ffb93c9c92951ffa8a6b663004932524f28ecf9c116a60"
+  license "Apache-2.0"
+
+  depends_on "go" => :build
+
+  def install
+    system "go", "build", "-ldflags",
+             "-s -w -X github.com/cloudskiff/driftctl/build.env=release
+             -X github.com/cloudskiff/driftctl/pkg/version.version=v#{version}",
+             *std_go_args
+  end
+
+  test do
+    assert_match "v#{version}", shell_output("#{bin}/driftctl version")
+    assert_match "0", shell_output("#{bin}/driftctl --no-version-check scan 2>&1", 1)
```

Comment:
We need a test that exercises the some of the functionality of the app. Version checks or usage checks (foo --version or foo --help) are not sufficient, as explained in the formula cookbook.

In most cases, a good test would involve running a simple test case: run #{bin}/foo input.txt.

- Then you can check that the output is as expected (with assert_equal or assert_match on the output of shell_output)
- You can also check that an output file was created, if that is expected: assert_predicate testpath/"output.txt", :exist?

Some advice for specific cases:

- If the formula is a library, compile and run some simple code that links against it. It could be taken from upstream's documentation / source examples.
- If the formula is for a GUI program, try to find some function that runs as command-line only, like a format conversion, reading or displaying a config file, etc.
- If the software cannot function without credentials, a test could be to try to connect with invalid credentials (or without credentials) and confirm that it fails as expected.
- Same if the software requires a virtual machine, docker instance, etc. to be running.

Reply:
The second test line 19 is here to check for an expected error due to credentials missing

Reply:
The only output in that case is `0`?

Reply:
No there's our usage and an invalid creds error message at the end. Wrong assertion, will change it with

```
assert_match /Invalid AWS Region/, shell_output("#{bin}/driftctl --no-version-check scan 2>&1", 1)
```

Will it be okay ?

Reply:
Changed in 309d809